### PR TITLE
Itinerary

### PR DESCRIPTION
**Pull request by Julien-laville**
Add direction basic function
The feature flag is set to false
is working :
- toggle direction panel
- suggest
- desktop road map (toggle detail)
- mobile road map
- geolocation
- switch route with the panel
- display path (there is a known bug with z-index)

### CODE REVIEW

File: config/default_config.yml [hunk 1]
```diff
@@ -52,4 +52,9 @@ performance:
...
 
 telemetry:
-  enabled: true
\ No newline at end of file
+  enabled: true
+
+direction:
+  enabled: false
+  service:
+    token: pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4M29iazA2Z2gycXA4N2pmbDZmangifQ.-g_vE53SD2WrJ6tFX7QHmA
```
Comment (amatissart) [inline config/default_config.yml]: to hide by default

File: src/adapters/map_direction.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  constructor(scene) {
+    this.scene = scene
+    listen('direction_add_layer', (options) => {
+      console.log(options)
+      this.scene.mb.addLayer(this.getGeoJSONPolygon(polygon, color))
+    })
+
+    listen('direction_fit_bounds', (bounds) => {
+
+    })
```
Comment (amatissart) [inline src/adapters/map_direction.js]: to clean up

File: src/views/app_panel.dot [hunk 3]
```diff
@@ -2,6 +2,7 @@
   <div class="favorite_poi_panel__container">
     {{= this.favoritePanel.render() }}
     {{= this.poiPanel.render() }}
+    {{= this.directionPanel.render() }}
```
Comment (amatissart) [inline src/views/app_panel.dot]: to render only if directions are enabled

File: src/views/service_panel.dot [hunk 4]
```diff
@@ -25,4 +25,8 @@
     <i class="icon-icon_star"></i>
   </button>
 
+  <button class="service_panel_mobile__direction" {{= click(this.toggleDirection, this) }}>
```
Comment (amatissart) [inline src/views/service_panel.dot]: to hide if directions are disabled

File: src/adapters/scene.js [hunk 5]
```diff
@@ -143,7 +148,7 @@ Scene.prototype.fitMap = function(poi, options = {}) {
     if(options.sidePanelOffset && window.innerWidth > layout.mobile.breakPoint) {
       padding.left += layout.sizes.panelWidth
     }
-    if(this.mb.getZoom() > MIN_ZOOM_FLYTO && this.isWindowedPoi(poi)) {
+    if(poi.getLngLat && this.mb.getZoom() > MIN_ZOOM_FLYTO && this.isWindowedPoi(poi)) {
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/adapters/scene.js]: fixed with a directionPoi

File: config/default_config.yml (hunk 1 above)
Comment (Julien-laville) [reply config/default_config.yml]: done

File: src/adapters/map_direction.js (hunk 2 above)
Comment (Julien-laville) [reply src/adapters/map_direction.js]: :)

File: src/views/app_panel.dot (hunk 3 above)
Comment (Julien-laville) [reply src/views/app_panel.dot]: sure

**Pull request merged**